src/main/webapp/app/course/manage/detail/course-detail-line-chart.component.ts

```diff
@@ -121,13 +126,18 @@ export class CourseDetailLineChartComponent implements OnChanges {
     }
 
     private createLabels() {
+        this.dataCopy[0].series = [{}];
+        this.absoluteSeries = [{}];
         const prefix = this.translateService.instant('calendar_week');
-        const startDate = dayjs().subtract(this.displayedNumberOfWeeks - 1 + this.displayedNumberOfWeeks * -this.currentPeriod, 'weeks');
-        const endDate = this.currentPeriod !== 0 ? dayjs().subtract(this.displayedNumberOfWeeks * -this.currentPeriod, 'weeks') : dayjs();
+        const endDate = dayjs().subtract(this.currentOffsetToEndDate + this.displayedNumberOfWeeks * -this.currentPeriod, 'weeks');
```

Comment:
`this.currentOffsetToEndDate + this.displayedNumberOfWeeks * -this.currentPeriod` appears in the calculation of the `startDate` again (with some other addition in between). Can you move it to a variable with a name that explains what is actually calculated here?

Reply:
Done in b9ecc6d